server/routes/banner.js

```diff
@@ -38,42 +61,41 @@ router.delete("/:id", ensureAuthenticated, (req, res) => {
 /**
  *  Returns a random banner given the current date
  */
-router.get("/random", (req, res) => {
+router.get("/random", async (req, res) => {
+
+  const bannersList = await bannerCtrl.getBannersArray();
 
-  const bannersList = bannerCtrl.getBannersArray();
   if ( bannersList.length > 0 ) {
+
     bannerCtrl
       .getRandomBanner(req.query)
       .then(imgPath => {
-        if (req.query.noredirect) {
-          const absolutePath = path.join(PUBLIC_PATH, imgPath);
-          res.sendFile(absolutePath);
-        } else {
-          res.redirect(imgPath);
-        }
+        res.redirect(imgPath);
       })
       .catch(e => {
         console.error(e);
         res.status(500).send(e.message);
       });
+
   } else {
     res.sendStatus(404);
   }
-
 });
 
 /**
  *  Handles banner picture upload
  */
 router.post("/upload", ensureAuthenticated, (req, res) => {
-  const bannerFolder = `${PUBLIC_PATH}assets/banners/`;
+
+  const bannerFolder = `assets/banners/`;
   req.pipe(req.busboy);
 
   req.busboy.on("file", (fieldname, file, filename) => {
 
     bannerCtrl
       .uploadBanner(bannerFolder, fieldname, file, filename)
-      .then(newFilename => res.send({data: newFilename}));
+      .then(newFilename => res.send({data: newFilename}))
+      .catch(e => console.log('e', e));
```

Comment:
Renvoyer une jolie `500` en plus du console log

Reply:
Done in 598bac694a5bc952b4ce5c34b0382652cc9884e1